# Neuronal Activation of the Gastrointestinal Tract Shapes the Gut Environment in Mice
####  Bryan B. Yoo, Jessica A. Griffiths, Peter Thuy-Boun, Victor Cantu, Kelly Weldon, Collin Challis, Michael J. Sweredoski, Ken Y. Chan, Taren M. Thron, Gil Sharon, Annie Moradian, Gregory Humphrey, Qiyun Zhu, Justin Shaffer, Dennis W. Wolan, Pieter C. Dorrestein, Rob Knight, Viviana Gradinaru, Sarkis K. Mazmanian

## Overview

This repository contains open-access data from the paper. 

## Summary

The enteric nervous system (ENS) coordinates responses that shape the intestinal milieu, and participates in bi-directional communication with various organs including the brain. Tools to study, and knowledge of, ENS development and function lag behind brain research. Herein, we deploy recombinant adeno-associated viral (rAAV) vectors with enhanced tropism for the gut to map and activate gut associated neurons in mice with spatial and temporal resolution. rAAV-mediated fluorescent labelling coupled with whole-tissue clearing methods in the small intestine and colon reveal novel neuronal architectures in the gut of mice. Viral delivery of genetically-encoded calcium indicators enable in vivo measurement of ENS activity. Further, we employed chemogentics to specifically activate gut neurons that express choline acetyltransferase (ChAT+) or tyrosine hydroxylase (TH+). Targeted activation of ChAT+ or TH+ neuronal populations associated with the gastrointestinal (GI) tract alters the intestinal transcriptome, mouse and microbial proteomes, metagenome, and fecal metabolome. We reveal previously unknown and broad roles for enteric neurons in modulating GI physiology, mucosal immunity, and microbiome structure, and provide evidence for novel functions by the ENS such as regulating fungal colonization and shaping of bile acid profiles in the gut. These advanced experimental tools and rich datasets with multi-parameter characterization of the gut ecosystem may facilitate expanded understanding and study of the ENS.


## Contents
1. [RNAseq data](https://github.com/jessicagriffiths/Yoo_et_al_2021/tree/main/RNAseq)
2. [Metagenomics data](https://github.com/jessicagriffiths/Yoo_et_al_2021/tree/main/metagenomics)
3. [Metabolomics data](https://github.com/jessicagriffiths/Yoo_et_al_2021/tree/main/metabolomics)
4. [Proteomics data](https://github.com/jessicagriffiths/Yoo_et_al_2021/tree/main/proteomics)
5. [GCaMP data](https://github.com/jessicagriffiths/Yoo_et_al_2021/tree/main/gcamp6f)
6. [ENS quantification](https://github.com/jessicagriffiths/Yoo_et_al_2021/tree/main/ENS%20quantification)

## Acknowledgements

We thank members in the Mazmanian’s laboratory for discussions throughout the research and critical reading of the manuscript. We thank Dr. Andres Collazo and Caltech’s Biological Imaging Facility for training access to microscopy capabilities and the Caltech Proteome Exploration Laboratory for access to LC-MS/MS capabilities. This work was supported by grants to S.K.M. from the Heritage Medical Research Institute, Emerald Foundation, Caltech Center for Environmental and Microbial Interactions (CEMI), Aligning Science Across Parkinson’s Foundation, the National Institutes of Health (GM007616 and DK078938), and the Department of Defense (PD160030).

## Contact
Corresponding Author. Email: byoo@caltech.edu (B.B.Y); sarkis@caltech.edu (S.K.M)


