# Peripheral Neuronal Activation of the Intestinal Tract Shapes the Microbiome and Alters Gut Physiology
####  Bryan B. Yoo*, Jessica A. Griffiths*, Peter Thuy-Boun, Victor Cantu, Kelly Weldon, Collin Challis, Michael J. Sweredoski, Ken Y. Chan, Taren M. Thron, Gil Sharon, Annie Moradian, Gregory Humphrey, Qiyun Zhu, Justin Shaffer, Dennis W. Wolan, Pieter C. Dorrestein, Rob Knight, Viviana Gradinaru, Sarkis K. Mazmanian

## Overview

This repository contains open-access data from the paper. 

## Summary

The gastrointestinal (GI) tract is extensively innervated by both extrinsic and intrinsic neurons of the enteric nervous system (ENS) that together regulate gut motility, secretion, and immunity. The GI tract also harbors a diverse microbiome, interactions between the ENS and the gut microbes remain poorly understood. Herein, we activated gut-associated neurons in mice to determine effects on intestinal microbial communities and their metabolites, as well as on host physiology. We used recombinant adeno-associated viral vectors with enhanced tropism for the gut, and no targeting to the bran, to chemogenetically activate either choline acetyltransferase (ChAT)-expressing or tyrosine hydroxylase (TH)-expressing neurons in the periphery. Targeted activation of discrete neuronal subtypes distinctively altered the metagenome, fecal metabolome, and mouse and microbial proteomes. We reveal broad and previously unknown roles for ChAT+ and TH+ neurons in modulating microbiome structure, and provide evidence for novel ENS functions such as shaping bile acid profiles and regulating fungal colonization of the gut. Further, ChAT+ neuronal activation upregulated transcriptional pathways for muscle cell proliferation, angiogenesis, and muscle development. While mice displayed increased fecal production following activation of gut-associated ChAT+ and TH+ neurons, only ChAT+ neuronal activation resulted in increased colonic migrating motor complexes and diarrhea-like fluid secretion. These findings suggest that specific subsets of peripherally-activated ENS neurons differentially regulate the gut microbiome and GI physiology in mice.
 


## Contents
1. [ENS quantification](https://github.com/jessicagriffiths/Yoo_et_al_2021/tree/main/ENS%20quantification)
2. [Metagenomics data](https://github.com/jessicagriffiths/Yoo_et_al_2021/tree/main/metagenomics)
3. [Metabolomics data](https://github.com/jessicagriffiths/Yoo_et_al_2021/tree/main/metabolomics)
4. [Proteomics data](https://github.com/jessicagriffiths/Yoo_et_al_2021/tree/main/proteomics)
5. [RNAseq data](https://github.com/jessicagriffiths/Yoo_et_al_2021/tree/main/RNAseq)



## Acknowledgements

We thank members in the Mazmanian’s laboratory for discussions throughout the research and critical reading of the manuscript. We thank Dr. Andres Collazo and Caltech’s Biological Imaging Facility for training access to microscopy capabilities and the Caltech Proteome Exploration Laboratory for access to LC-MS/MS capabilities. This research was funded in whole or in part by Aligning Science Across Parkinson’s ASAP-000375 through the Michael J. Fox Foundation for Parkinson’s Research (MJFF). For the purpose of open access, the author has applied a CC BY 4.0 public copyright license to all Author Accepted Manuscripts arising from this submission. This work was supported by grants to S.K.M. from the Heritage Medical Research Institute, Emerald Foundation, Caltech Center for Environmental and Microbial Interactions (CEMI), the National Institutes of Health (GM007616 and DK078938), and the Department of Defense (PD160030).

## Contact
Corresponding author email: sarkis@caltech.edu (S.K.M)

* indicates co-first authors
