We profiled gene expression with QuantSeq, a quantitative 3â€² mRNA sequencing technology, in 1 cm of tissue from the distal SI and proximal colon harvested 1 h after the last C21 injection.  De-multiplexed sequencing reads were generated using Illumina BaseSpace. To analyze the reads, the umi2index tool was used to add the 6 nucleotide UMI sequence to the identifier of each read and trim the UMI from the start of each read. This generated a new FASTQ file, which was then processed through trimming and alignment. Second, after the quality and polyA trimming by BBDuk and alignment by HISAT2, the mapped reads were collapsed according to the UMI sequence of each read. Reads were collapsed if they had the same mapping coordinates (CIGAR string) and identical UMI sequences. Collapsing reads in this manner removes PCR duplicates. Read counts were calculated using HTSeq by supplementing Ensembl gene annotation (GRCm38.78). Raw read counts were run through ShinySeq to obtain differentially expressed genes and downstream gene ontology analyses.
